ext/intl/converter/converter.c

```diff
@@ -105,18 +106,14 @@ static void php_converter_default_callback(zval *return_value, zval *zobj, zend_
 /* {{{ proto void UConverter::toUCallback(int $reason,
                                           string $source, string $codeUnits,
                                           int &$error) */
-ZEND_BEGIN_ARG_INFO_EX(php_converter_toUCallback_arginfo, 0, ZEND_RETURN_VALUE, 4)
-	ZEND_ARG_INFO(0, reason)
-	ZEND_ARG_INFO(0, source)
-	ZEND_ARG_INFO(0, codeUnits)
-	ZEND_ARG_INFO(1, error)
-ZEND_END_ARG_INFO();
 static PHP_METHOD(UConverter, toUCallback) {
 	zend_long reason;
-	zval *source, *codeUnits, *error;
+	char *source, *codeUnits;
+	size_t source_len, codeUnitsLen;
+	zval *error;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "lzzz",
-		&reason, &source, &codeUnits, &error) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "lssz",
+		&reason, &source, &source_len, &codeUnits, &codeUnitsLen, &error) == FAILURE) {
```

Comment:
Can use `S` and `zend_string*` to save some variables.

Reply:
Ah, I see. Actually, I wasn't sure which one is the preferred. I checked that `s` is used more more often so I settled on it.